sw-emulator/lib/cpu/src/csr_file.rs

```diff
@@ -165,10 +181,45 @@ impl CsrFile {
     pub fn write(&mut self, addr: RvAddr, val: RvData) -> Result<(), RvException> {
         let addr = addr as usize;
         const CSR_MAX: usize = CsrFile::CSR_COUNT - 1;
+        const MSTATUS_MIE: u32 = 1 << 3;
+        const MIE_MEI: u32 = 1 << 11;
+        const MPMC_HALT: u32 = 1 << 0;
+        const MPMC_MIE: u32 = 1 << 1;
         match addr {
             0..=CSR_MAX => {
                 let csr = &mut self.csrs[addr];
                 csr.val = (csr.val & !csr.mask) | (val & csr.mask);
+
+                if addr == Csr::MEIVT as usize {
+                    self.timer
+                        .schedule_action_in(0, TimerAction::SetExtIntVec { addr: csr.val });
+                }
+                if addr == Csr::MSTATUS as usize {
+                    self.timer.schedule_action_in(
+                        0,
+                        TimerAction::SetGlobalIntEn {
+                            en: val & MSTATUS_MIE != 0,
+                        },
+                    );
+                    // Let's see if the soc wants to interrupt
+                    self.timer.schedule_poll_in(2);
```

Comment:
Yes, this is the best way to do this. The fact that we can delay it a few cycles is nice because it gives us the option to make things behave more like the real hardware.